packages/flutter_tools/lib/src/resident_runner.dart

```diff
@@ -78,6 +77,29 @@ abstract class ResidentRunner {
 
   bool get supportsRestart => false;
 
+  String isolateFilter;
+
+  List<FlutterView> _currentViewsCache;
+  List<FlutterView> get currentViews {
+    if (_currentViewsCache == null) {
+      if ((vmServices == null) || vmServices.isEmpty) {
+        return null;
+      }
+      if (isolateFilter == null) {
+        return vmServices[0].vm.views.toList();
+      }
+      final List<FlutterView> result = <FlutterView>[];
+      for (VMService service in vmServices) {
+        result.addAll(service.vm.allViewsWithName(isolateFilter));
+      }
+      _currentViewsCache = result;
+    }
+    return _currentViewsCache;
+  }
+
+  FlutterView get currentView =>
```

Comment:
nit: style guide says not to put a newline in an expression containing `=>`. This should either be all on one line or use a block with `return`.